caddyconfig/httpcaddyfile/builtins.go

```diff
@@ -619,11 +619,46 @@ func parseHandleErrors(h Helper) ([]ConfigValue, error) {
 //     }
 //
 func parseLog(h Helper) ([]ConfigValue, error) {
+	return parseLogHelper(h, false, nil)
+}
+
+// parseLogHelper is used both for the parseLog directive within Server Blocks,
+// as well as the global "log" option for configuring loggers at the global
+// level. The globalOption parameter is used to distinguish any differing logic
+// between the two.
+func parseLogHelper(h Helper, globalOption bool, globalLogNames map[string]struct{}) ([]ConfigValue, error) {
```

Comment:
Will `globalOption` ever be `true` while `globalLogNames` is nil? If not, we can drop the `globalOption` param and just assume that if `globalLogNames` is non-nil, `globalOption` must be true.

Reply:
Good point! I think that was an artifact of an earlier iteration, I collapsed the parameters and added an internal variable to keep the logic explicit.